From 8c47202f6b5556f92f01dada12fd7b9a278ee164 Mon Sep 17 00:00:00 2001
From: Dario Faggioli <raistlin@linux.it>
Date: Wed, 11 Apr 2012 11:19:53 +0200
Subject: [PATCH 05/48] sched: SCHED_DEADLINE avg_update accounting.

Make the core scheduler and load balancer aware of the load
produced by -deadline tasks, by updating the moving average
like for sched_rt.

Signed-off-by: Dario Faggioli <raistlin@linux.it>
Signed-off-by: Juri Lelli <juri.lelli@gmail.com>
---
 kernel/sched/dl.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/sched/dl.c b/kernel/sched/dl.c
index fc13a45..32ca118 100644
--- a/kernel/sched/dl.c
+++ b/kernel/sched/dl.c
@@ -524,6 +524,8 @@ static void update_curr_dl(struct rq *rq)
 	curr->se.exec_start = rq->clock;
 	cpuacct_charge(curr, delta_exec);
 
+	sched_rt_avg_update(rq, delta_exec);
+
 	dl_se->runtime -= delta_exec;
 	if (dl_runtime_exceeded(rq, dl_se)) {
 		__dequeue_task_dl(rq, curr, 0);
-- 
1.7.9.5

